db.c

```diff
@@ -350,6 +350,7 @@ Cursor* leaf_node_find(Table* table, uint32_t page_num, uint32_t key) {
   Cursor* cursor = malloc(sizeof(Cursor));
   cursor->table = table;
   cursor->page_num = page_num;
+	cursor->end_of_table = false;
```

Comment:
it looks like the indentation is a little off. Otherwise, looks good!

Reply:
Ok, I will fix it right away.